WPFHospitalEditor/EquipmentSearchResultDialog.xaml.cs

```diff
@@ -0,0 +1,260 @@
+﻿using HealthcareBase.Dto;
+using System;
+using System.Collections.Generic;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Media;
+using WPFHospitalEditor.Controller;
+using WPFHospitalEditor.MapObjectModel;
+using WPFHospitalEditor.Service;
+
+namespace WPFHospitalEditor
+{
+    /// <summary>
+    /// Interaction logic for EquipmentSearchResultDialog.xaml
+    /// </summary>
+    public partial class EquipmentSearchResultDialog : Window
+    {
+        List<Button> advanceSearchButtons = new List<Button>();
+        private Dictionary<int, MapObject> row = new Dictionary<int, MapObject>();
+        int firstContentRowNumber = 2;
+        IMapObjectController mapObjectController = new MapObjectController();
+        private HospitalMap hospitalMap;
+        private Role role;
+        ScrollViewer viewer = new ScrollViewer();
+
+        public EquipmentSearchResultDialog(HospitalMap hospitalMap, Role role)
+        {
+            this.Height = AllConstants.SearchDialogHeight;
+            InitializeComponent();
+            DefineDynamicGrid();
+            this.role = role;
+            this.hospitalMap = hospitalMap;
+        }
+
+        private void DefineDynamicGrid()
+        {
+            createRows();
+            createRowContent();
+            viewer.Content = DynamicGrid;
+            viewer.BorderBrush = Brushes.AliceBlue;
+            Border.Child = viewer;
+        }
```

Comment:
ScrollViewer should be defined in xaml, right?

Reply:
Now it is in xaml. :)